Symfony/CS/Tests/Fixer/Contrib/Php4ConstructorFixerTest.php

```diff
@@ -0,0 +1,467 @@
+{
+    /**
+     * PHP-5 Constructor
+     */
+    function __construct($bar)
+    {
+        $this->Foo($bar);
+    }
+
+    /**
+     * PHP-4 Constructor
+     */
+    function Foo($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo
+{
+
+    /**
+     * PHP-4 Constructor
+     */
+    function __construct($bar)
+    {
+        var_dump(1);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testPhp4Parent()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        parent::FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testPhp4ParentInit()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        parent::init(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::init(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testMixedParent()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $this->makeTest($expect, $input);
+    }
+
+    public function testParentOther()
+    {
+        $input = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function Foo($bar)
+    {
+        $this->FooParent(1);
+        var_dump(9);
+    }
+
+    function bar()
+    {
+        var_dump(3);
+    }
+}
+EOF;
+
+        $expect = <<<'EOF'
+<?php
+
+class Foo extends FooParent
+{
+    /**
+     * Constructor
+     */
+    function __construct($bar)
+    {
+        parent::__construct(1);
```

Comment:
So now, let's imagine that our `Foo` class extends some old external library `ExternalVendorLibraryInPhp4`.
And we call a parent constructor `parent::ExternalVendorLibraryInPhp4(1)`.
changing it into `parent::__construct(1)` as you proposed is great for our code, but you don't know if the class that we inherit from are out and will we fix it or no.
Even if they are inside one project Fixer may be run to fix only one file, and it will break the logic.
And what if constructor name can be fixed in `Foo` class, but can't be fixed in class that we extend?

Fixer itself propose a great rule, but it may break a code and possibly break it. I will not only leave it in CONTRIB namespace, but also mark as risky (`Warning! This could change code behavior.` in descr, see `strict` fixer).

//cc @stof